nemo/src/model/rule_model/numeric_literal.rs

```diff
@@ -11,6 +13,17 @@ pub enum NumericLiteral {
     Double(Double),
 }
 
+impl NumericLiteral {
+    /// Get primitive type that fits the literal
+    pub fn primitive_type(&self) -> PrimitiveType {
+        match self {
+            Self::Integer(_) => PrimitiveType::Integer,
+            Self::Double(_) => PrimitiveType::Float64,
+            Self::Decimal(_, _) => PrimitiveType::Float64, // TODO: is this what we want?
```

Comment:
Pretty sure this is not what we want, since `f64` is not a faithful representation of `Decimal`. In particular, there are integers in `u64` (and hence, also in `Decimal`) that are not representable in `f64`.

Reply:
True, this should probably fall back to `Any` then...